modin/core/dataframe/algebra/binary.py

```diff
@@ -196,19 +300,24 @@ def caller(
                 Result of binary function.
             """
             axis = kwargs.get("axis", 0)
+            if isinstance(other, type(query_compiler)) and broadcast:
+                assert (
+                    len(other.columns) == 1
+                ), "Invalid broadcast argument for `broadcast_apply`, too many columns: {}".format(
+                    len(other.columns)
+                )
+                # Transpose on `axis=1` because we always represent an individual
+                # column or row as a single-column Modin DataFrame
+                if axis == 1:
+                    other = other.transpose()
+            if dtypes != "copy":
+                dtypes = try_compute_new_dtypes(
+                    query_compiler, other, infer_dtypes, dtypes
+                )
+
             shape_hint = None
             if isinstance(other, type(query_compiler)):
                 if broadcast:
-                    assert (
```

Comment:
this was moved upper so `try_compute_new_dtypes` would receive properly transposed `other`